devito/archinfo.py

```diff
@@ -187,24 +187,32 @@ def simd_items_per_reg(self, dtype):
 
 class Cpu64(Platform):
 
-    def _detect_isa(self):
-        return 'cpp'
-
-
-class Intel64(Cpu64):
+    # The known isas ​​will be overwritten in the specialized classes
+    known_isas = tuple()
 
     def _detect_isa(self):
-        known_isas = ['cpp', 'sse', 'avx', 'avx2', 'avx512']
-        for i in reversed(known_isas):
+        for i in reversed(self.known_isas):
             if any(j.startswith(i) for j in get_cpu_info()['flags']):
                 # Using `startswith`, rather than `==`, as a flag such as 'avx512'
                 # appears as 'avx512f, avx512cd, ...'
                 return i
         return 'cpp'
 
 
+class Intel64(Cpu64):
+
+    known_isas = ('cpp', 'sse', 'avx', 'avx2', 'avx512')
+
+    def _detect_isa(self):
+        return super(Intel64, self)._detect_isa()
+
+
 class Arm(Cpu64):
-    pass
+
+    known_isas = ('fp', 'asimd', 'asimdrdm')
+
+    def _detect_isa(self):
```

Comment:
you can drop this. It's gonna automatically pick the one of the superclass

Reply:
Oh, that's right. I'm gonna do that.